base/uk.ac.stfc.isis.ibex.dae/src/uk/ac/stfc/isis/ibex/dae/spectra/Spectrum.java

```diff
@@ -19,44 +19,128 @@
 
+/**
+ * The view model for a single spectrum plot.
+ */
 public class Spectrum extends ModelObject {
 
 	private int number;
 	private int period;
 	private double[] xData = new double[2];
 	private double[] yData = new double[2];
 	
+	private int spectrumYAxisTypeSelectionIndex = 0;
+	private Preferences preferences;
+	
+	/**
+	 * Constructor.
+	 * @param preferenceStore the preference store to use.
+	 */
+	public Spectrum(Preferences preferenceStore) {
+		this.preferences = preferenceStore;
+		setNumber(preferences.getInt("spectrumNumber", 1));
+		setPeriod(preferences.getInt("period", 1));
+		setTypeSelectionIndex(preferences.getInt("typeSelectionIndex", 0));
+	}
+	
+	/**
+	 * Gets the spectrum number.
+	 * @return the spectrum number
+	 */
 	public int getNumber() {
 		return number;
 	}
 	
+	/**
+	 * Sets the spectrum number.
+	 * @param value the new spectrum number
+	 */
 	public void setNumber(int value) {
 		firePropertyChange("number", number, number = value);
+		preferences.putInt("spectrumNumber", number);
 	}
 	
+	/**
+	 * Gets the period of this spectrum.
+	 * @return the period
+	 */
 	public int getPeriod() {
 		return period;
 	}
 	
+	/**
+	 * The period of this spectrum.
+	 * @param value the new period
+	 */
 	public void setPeriod(int value) {
 		firePropertyChange("period", period, period = value);
+		preferences.putInt("period", period);
 	}
 	
+	/**
+	 * The X data of this spectrum.
+	 * @return the X data.
+	 */
 	public double[] xData() {
 		return xData;
 	}
 		
+	/**
+	 * The Y data of this spectrum.
+	 * @return the Y data.
+	 */
 	public double[] yData() {
 		return yData;
 	}
 	
-	protected void setXData(double[] value) {
+	/**
+	 * Sets the selection index of the Y axis type.
+	 * 
+	 * This is an index into the {@link SpectrumYAxisTypes} enum.
+	 */
+	public void setTypeSelectionIndex(int type) {
+		firePropertyChange("typeSelectionIndex", spectrumYAxisTypeSelectionIndex, spectrumYAxisTypeSelectionIndex = type);
+		preferences.putInt("typeSelectionIndex", spectrumYAxisTypeSelectionIndex);
+	}
+
+	/**
+	 * Gets the selection index of the Y axis type.
+	 * 
+	 * This is an index into the {@link SpectrumYAxisTypes} enum.
+	 * 
+	 * @return the type selection index
+	 */
+	public int getTypeSelectionIndex() {
+		return spectrumYAxisTypeSelectionIndex;
+	}
+
+
+	/**
+	 * Sets the X data array for this spectrum.
+	 * @param value the data to set.
+	 */
+	public void setXData(double[] value) {
 		firePropertyChange("xData", xData, xData = value);
 	}
 	
-	protected void setYData(double[] value) {
+	/**
+	 * Sets the Y data array for this spectrum.
+	 * @param value the data to set.
+	 */
+	public void setYData(double[] value) {
+		if (SpectrumYAxisTypes.values()[spectrumYAxisTypeSelectionIndex] == SpectrumYAxisTypes.ABSOLUTE_COUNTS) {
+			for (int i = 0; i<value.length; i++) {
+				try {
+					value[i] *= Math.abs(xData[i+1] - xData[i]);
+				} catch (ArrayIndexOutOfBoundsException e){
+					value[i] = 0;
+				}
```

Comment:
There are two possible exception sceanrios here:

- `xData` isn't as long as you expect
- `value[value.length-1]` can't be computed because both arrays have the same length but it requires `xData[value.lenth] which is out of range

These are two different scenarios which are not distinguished by the catch. The first scenario does sound exceptional and so a try-catch would be a reasonable control.

For the 2nd scenario, is there ever a case where there will be excess `xData`? If this is written instead as:

```
for (int i = 0; i<value.length-1; i++) {
        value[i] *= Math.abs(xData[i+1] - xData[i]);
value[i-1] = 0;
```

The code ends up being cleaner and it shows understanding is that there will always be one value we can't compute with this algorithm, rather than it being an exceptional, unpredictable event. In some cases it's better to ask for forgiveness than permission, but I think this in this case it obfuscates our understanding of what's going on.